website/utils/url_utils.ts

```diff
@@ -149,14 +164,138 @@ export function parseKeyValueList(data: string): Record<string, string> {
   return result;
 }
 
-function parseFloat(value: string | undefined): number | undefined {
+/**
+ * Parses a string to a float and clamps the result to the [min, max] range.
+ * Returns `undefined` if the string is undefined or NaN.
+ * @param value String to parse as a float. Will be parsed with `Number.parseFloat`.
+ * @param min Minimum value, inclusive.
+ * @param max Maximum value, inclusive.
+ * @returns
+ * - The parsed number, clamped to the [min, max] range.
+ * - `undefined` if the string is undefined or NaN.
+ */
+export function parseStringFloat(value: string | undefined, min: number, max: number): number | undefined {
   if (value === undefined) {
     return undefined;
   }
   const number = Number.parseFloat(value);
-  return Number.isNaN(number) ? undefined : number;
+  return Number.isNaN(number) ? undefined : clamp(number, min, max);
+}
+
+/**
+ * Parses a string to an integer and clamps the result to the [min, max] range.
+ * @param value String to parse as a float. Assumes base 10, parses with `Number.parseInt(value, 10)`.
+ * @param min Minimum value, inclusive.
+ * @param max Maximum value, inclusive.
+ * @returns
+ * - The parsed number, clamped to the [min, max] range.
+ * - `undefined` if the string is undefined or NaN.
+ */
+export function parseStringInt(value: string | undefined, min: number, max: number): number | undefined {
+  if (value === undefined) {
+    return undefined;
+  }
+  const number = Number.parseInt(value, 10);
+  if (Number.isNaN(number)) {
+    return undefined;
+  }
+  return clamp(number, min, max);
+}
+
+/**
+ * Parses a string to an enum value; if the string is not in the enum, returns the default value.
+ * @param value String to parse.
+ * @param enumValues Enum. Cannot be a `const enum`, as these are removed at compile time.
+ * @param defaultValue Default value to return if the string is not in the enum.
+ * @returns A value from the enum or the default value. Note that the return type includes `undefined`
+ * if the `defaultValue` is `undefined`.
+ */
+export function parseStringEnum<E extends string, T extends E | undefined>(
+  value: string | undefined,
+  enumValues: Record<string | number | symbol, E>,
+  defaultValue: T = undefined as T
+): T {
+  if (value === undefined || !Object.values(enumValues).includes(value as E)) {
+    return defaultValue;
+  }
+  return value as T;
+}
+
+/**
+ * Parses a string boolean value ("1" as true, "0" as false), and returns `undefined` if the value is `undefined`.
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
All of these `eslint-disable` statements will be removed in the next PR. These also denote the methods I did not write unit tests for, but will be tested via their callees in the next PR.